### ISSUE TITLE
More descriptive output in VMWare and Commserve plugins

### ISSUE THREAD

**Issue by jQubz** [centreon/centreon-plugins#2971]
I would like to propose an improvement mainly for VMWare and Commserve plugins.

Let’s take host-alarm in VMWare vCenter plugin as an example first. Right now a critical alert returns output of “CRITICAL: 1 problem(s) detected”. This is a correct indication of a problem, but doesn't really immediately direct a user to what the issue may be. Extended output offers a lot more info in this case, but it has the opposite problem - it's too descriptive, because it marks every action it takes leaving a lot of line like “checking host 'XXXX’”. Separate problem is that extended output isn't logged, so it doesn't leave easy to read history of what exactly was detected by monitoring.

My proposal is to introduce a new flag (for example --extend-output) that could be used to make the short output a little bit more clear by filtering the message from extended output that was the cause of the problem. This is done already in different modes- keeping to the example of VMWare plugin, mode status-host presents output like “CRITICAL: Host 'XXXX' : overall status is red” which is the alerting string filtered straight from the extended output and exactly how I imagine the enhancment to work.

Apart from the VMWare vCenter plugin I also noticed similiar behavior in case of Commvault CommServe Rest API plugin, I would love to have this enhancement introduced there too.

Justification for such improvement is to provide monitoring users with clear indication of the problem from the start. Correlation of events from monitoring and logs is sometimes hard, so root cause of serious events may be needlessly missed.

**Comment by garnier-quentin** [comment]
You have the detailed information in the extended output (with verbose). If we put that in short output, it won't be short anymore ;)

**Comment by jQubz** [comment]
Hi :)

That is totally true, but that's also the reason I proposed it as a flag which would allow everyone to choose if they want to have this in their output. Also, I would argue that in a lot of cases if you filter just the important string (the one that was the cause of the alert) and it's still a lot of text then it's sensible to include it in the output, as it's especially important then. This is done anyway in modes like status-host where if you have multiple ESXi hosts down the output is getting looooong.
Also, is there a way to log the extended output? One of the major problems we encountered is that there is absolutely no way to check what each alert meant as the only info we are getting in Centreon logs is "X problem(s) detected".
